app/controllers/assignments_controller.rb

```diff
@@ -208,16 +208,14 @@ def index
       @section = current_user.section
       # get results for assignments for the current user
       @a_id_results = Hash.new()
-      @assignments.each do |a|
-        if current_user.has_accepted_grouping_for?(a)
-          grouping = current_user.accepted_grouping_for(a)
-          if grouping.has_submission?
-            submission = grouping.current_submission_used
-            if submission.has_remark? && submission.remark_result.released_to_students
-              @a_id_results[a.id] = submission.remark_result
-            elsif submission.has_result? && submission.get_original_result.released_to_students
-              @a_id_results[a.id] = submission.get_original_result
-            end
+      accepted_groupings = current_user.accepted_groupings.includes(:assignment, :current_submission_used)
+      accepted_groupings.each do |grouping|
+        if grouping.has_submission?
```

Comment:
I realized because of this change, there should be an extra check here for `!grouping.assignment.id_hidden` (if the assignment is hidden, this result should not be shown to students yet).